changelog/7259.deprecation.rst

```diff
@@ -1,3 +1,7 @@
 ``py.path.local`` arguments for hooks have been deprecated. See :ref:`the deprecation note <legacy-path-hooks-deprecated>` for full details.
 
 ``py.path.local`` arguments to Node constructors have been deprecated. See :ref:`the deprecation note <node-ctor-fspath-deprecation>` for full details.
+
+.. note::
+    The name of the ``Node`` arguments and attributes (old ``fspath``, new ``path``) is **the opposite**
+    of the situation for hooks (old ``path``, new ``fspath``).
```

Comment:
I'm sure many people will ask the question "What, why??" when reading this... it might save us some explaining if we add a quick explanation, perhaps something along the lines: 

```suggestion
    of the situation for hooks (old ``path``, new ``fspath``). This is an unfortunate artifact due to historical reasons,
    which should be resolved in future versions as we slowly get rid of the :pypi:`py` dependency (see :issue:`9283` for a longer discussion).
```